README.md

```diff
@@ -5,754 +5,864 @@
+  { name: 'Watch', category: 'accessory' }
+]
 
-Composed states have two default transitions `set` and `merge`.
+import { create } from 'microstates';
 
-`set` will replace the state of the current microstate.
+let shop = create(Shop, { products, filter });
 
-```js
-create(Person).father.father.set({ name: 'Peter' }).state;
-// => { name: '', father: { name: 'Peter' }}
+shop.valueOf();
+// give you the value that can be used to restore the state
 ```
 
-`merge` will recursively merge the object into the current state.
+Microstates actually does deserialization by default. When you create a Microstate object, you provide the `create` function with the type and a POJO that represents the value. Microstates internally will create all of the instances using the type as a blueprint for the final state. 
 
-```js
-create(Person, { name: 'Peter' }).merge({ name: 'Taras', father: { name: 'Serge' } }).state;
-// { name: 'Taras', father: { name: 'Serge' }}
-```
+As discussed earlier, Microstates methods return copies that are derived from original microstate. These copies will have a new value. This allows to serialize/deserialize a type to any derived state. 
 
-## Static values
+For those familiar with tools like ReactDevTools and time travel debugging, Microstates makes it possible to use these kinds of tools on individual Microstate level. Microstates can work like tiny Redux stores where you don't have to write reducers while still get the benefits of time travel debugging.
 
-You can define static values on custom types. Static values are added to state but do not get
-transitions.
+In summary, Microstates are immutable JavaScript objects that are serializable, allow you to cache getters and provide a mechanism to declaratively derive next state using class like instance methods. They look like regular JavaScript classes but they hold conventions and performance improvements for state composed of many smaller states.
 
-```js
-class Ajax {
-  content = null;
-  isLoaded = false;
-}
+## Why Microstates?
 
-create(Ajax).state.content;
-// => null;
+Out tools effect how we solve problems and collaborate. Two tools can serve the same purpose but foster a completely different kind of ecosystem. Take jQuery plugins and React components as an example. Both of these tools provided a way to add custom behaviour to a DOM element but they fostered very different communities. 
 
-create(Ajax).content;
-// undefined
+In React world today, we see many special purpose components that are easy to combine to create sophisticated user experiences. These components tends to have few options but provide just the right API for you to build what you need. 
 
-create(Ajax).state.isLoaded;
-// => false
+jQuery plugins on the other hand offer endsless list of options that are difficult get just right. In jQuery ecosystem, libraries tended to be monolithic because plugins from different libraries often did not work well together. 
```

Comment:
endsless list of options -> endless options
In **the** jQuery ecosystem
often **do** not work well together.